freestyle/shared/src/main/scala/freestyle/free.scala

```diff
@@ -4,6 +4,10 @@ import scala.annotation.{compileTimeOnly, StaticAnnotation}
 import scala.language.experimental.macros
 import scala.reflect.macros.blackbox
 
+trait FreeModuleLike {
```

Comment:
¿Do you mean that the `@free` macro should also rewrite the `@free`-annotated trait, so as to add `Algebra` to its subclasses? ¿Or do you just mean to rename the trait?